agents_infra/agents_infra/utils/network.py

```diff
@@ -0,0 +1,61 @@
+import json
+import socket
+
+import urllib2
+from logtools import maybe_log_message
+from urlparse import urljoin, urlparse
+
+
+def is_tcp_reachable(url, timeout=3):
+    parsed = urlparse(url)
+    host = parsed.hostname
+    port = parsed.port or (443 if parsed.scheme == 'https' else 80)
+
+    try:
+        sock = socket.create_connection((host, port), timeout)
+        sock.close()
+        return True
+    except socket.error:
+        return False
+
+
+def check_http_health(url, api_key=None, auth_token_type=None, timeout=3):
```

Comment:
- Add a docstring, e.g., "Performs a GET /health request to the URL, returning True if status is healthy."
- Optionally allow the health endpoint to be configurable.